src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallCommand.cs

```diff
@@ -64,67 +66,97 @@ private void ValidateWorkloadIdsInput()
+                    }
 
-                var packageUrls = GetPackageDownloadUrlsAsync(workloadsToDownload, _skipManifestUpdate, _includePreviews).GetAwaiter().GetResult();
+                    recorder.HistoryRecord.RollbackFileContents = rollbackContents;
+                }
 
-                Reporter.WriteLine("==allPackageLinksJsonOutputStart==");
-                Reporter.WriteLine(JsonSerializer.Serialize(packageUrls, new JsonSerializerOptions() { WriteIndented = true }));
-                Reporter.WriteLine("==allPackageLinksJsonOutputEnd==");
-            }
-            else if (!string.IsNullOrWhiteSpace(_downloadToCacheOption))
-            {
-                try
+                if (_printDownloadLinkOnly)
                 {
+                    Reporter.WriteLine(string.Format(LocalizableStrings.ResolvingPackageUrls, string.Join(", ", _workloadIds)));
+
                     //  Take the union of the currently installed workloads and the ones that are being requested.  This is so that if there are updates to the manifests
                     //  which require new packs for currently installed workloads, those packs will be downloaded.
                     //  If the packs are already installed, they won't be included in the results
                     var existingWorkloads = GetInstalledWorkloads(false);
                     var workloadsToDownload = existingWorkloads.Union(_workloadIds.Select(id => new WorkloadId(id))).ToList();
 
-                    DownloadToOfflineCacheAsync(workloadsToDownload, new DirectoryPath(_downloadToCacheOption), _skipManifestUpdate, _includePreviews).Wait();
+                    var packageUrls = GetPackageDownloadUrlsAsync(workloadsToDownload, _skipManifestUpdate, _includePreviews).GetAwaiter().GetResult();
+                    PrintDownloadLink(packageUrls);
                 }
-                catch (Exception e)
+                else if (!string.IsNullOrWhiteSpace(_downloadToCacheOption))
                 {
-                    throw new GracefulException(string.Format(LocalizableStrings.WorkloadCacheDownloadFailed, e.Message), e, isUserError: false);
+                    try
+                    {
+                        //  Take the union of the currently installed workloads and the ones that are being requested.  This is so that if there are updates to the manifests
+                        //  which require new packs for currently installed workloads, those packs will be downloaded.
+                        //  If the packs are already installed, they won't be included in the results
+                        var existingWorkloads = GetInstalledWorkloads(false);
+                        var workloadsToDownload = existingWorkloads.Union(_workloadIds.Select(id => new WorkloadId(id))).ToList();
+
+                        recorder.Run(() => DownloadToOfflineCacheAsync(workloadsToDownload, new DirectoryPath(_downloadToCacheOption), _skipManifestUpdate, _includePreviews).Wait());
```

Comment:
I'm pretty sure `DownloadToOfflineCacheAsync` doesn't modify the installed workload versions or workloads, so we probably don't need to write a workload history record in this case.